lib/synapse/service_watcher/zookeeper_dns.rb

```diff
@@ -51,8 +51,9 @@ class Dns < Synapse::ServiceWatcher::DnsWatcher
       # Overrides the discovery_servers method on the parent class
       attr_accessor :discovery_servers
 
-      def initialize(opts={}, synapse, message_queue)
+      def initialize(opts={}, parent=nil, synapse, message_queue)
```

Comment:
is this instantiated anywhere other than your change below? if not, why the need for the default arg?